documentation/source/regulator.rst

```diff
@@ -0,0 +1,111 @@
+##################################
+The cocotb ``regulator`` Testbench
+##################################
+
+.. versionadded:: 1.5
+
+This is the testbench :mod:`test_regulator` for the design ``regulator`` showing
+how cocotb can be used in an analog-mixed signal (AMS) simulation.
+
+******************
+Overview Schematic
+******************
+
+.. image:: regulator_schematic.png
+
+
+**********
+The Design
+**********
+
+The design ``i_regulator`` consists of a trimmable regulator model written in Verilog-AMS
+(instance name ``i_regulator_block``), and load resistor (instance name ``i_resistor``).
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/regulator.sv
+   :caption: regulator.sv
+   :language: systemverilog
+   :start-at: module regulator
+
+The regulator block itself is modelled as shown below:
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/regulator_block.vams
+   :caption: regulator_block.vams
+   :language: systemverilog
+   :start-at: `include "disciplines.vams"
+
+
+*************
+The Testbench
+*************
+
+The testbench consists of both an :term:`HDL` part and a Python/cocotb part.
+While having :term:`HDL` as part of the testbench is not common with cocotb, it is perfectly possible.
+
+The HDL part of the Testbench
+=============================
+
+The testbench :term:`HDL` part is written in SystemVerilog and instantiates the design described above
+as ``i_regulator``.
+It also contains a probe module for analog values as instance ``i_analog_probe`` —
+imagine this being a multimeter that you quickly connect to different nodes in the design,
+measuring either voltage or current.
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/tb_regulator.sv
+   :caption: tb_regulator.sv
+   :language: systemverilog
+   :start-at: import nettypes_pkg
+
+The probe module can capture voltage and current of a node specified by ``node_to_probe``
+(a string in this module containing a hierarchical path).
+The capturing occurs whenever there is an edge on the logic signals
+``probe_voltage_toggle`` or ``probe_current_toggle``.
+The captured values can be read on real-value signals ``voltage`` and ``current`` in this module.
+
+Here is the capture code for ``voltage`` with the "user-interface" highlighted:
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/analog_probe_cadence.sv
+   :caption: analog_probe_cadence.sv
+   :language: systemverilog
+   :start-at: var string node_to_probe
+   :end-at: end  // probe_voltage
+   :emphasize-lines: 1-4
+   :dedent: 2
+
+.. note:: This analog probe module is currently only implemented for the
+          Cadence Incisive and Xcelium simulators (with the AMS option available).
```

Comment:
This is also out of date. It also works with Synopsys VCS, correct?

Reply:
Actually, no, it doesn't. I have added stubs with some information about VCS-MX that I was able to find, but couldn't try it for lack of a  Synopsys analog simulator available.

Reply:
Ah, got it.